apps/explorer/src/hooks/useSearch.ts

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import {
+    isValidTransactionDigest,
+    type SuiTransactionResponse,
+    isValidSuiAddress,
+    isValidSuiObjectId,
+    normalizeSuiObjectId,
+    is,
+    SuiObject,
+} from '@mysten/sui.js';
+import { useQuery } from '@tanstack/react-query';
+
+import { useRpc } from '~/hooks/useRpc';
+import { isGenesisLibAddress } from '~/utils/api/searchUtil';
+
+export function useSearch(query: string) {
+    const rpc = useRpc();
+
+    return useQuery(
+        ['search', query],
+        () => {
+            const promises = [];
```

Comment:
I'd actually type this as something like `Promise<{label: string, results: something[]|>` just to ensure you're pushing the right thing to this